ts/client/src/accounts/oracle.ts

```diff
@@ -186,3 +236,50 @@ export function isOracleStaleOrUnconfident(
 
   return false;
 }
+
+export function deriveFallbackOracleQuoteKey(
+  accountInfo: AccountInfo<Buffer>,
+): PublicKey {
+  if (isOrcaOracle(accountInfo)) {
+    const tokenA = new PublicKey(accountInfo.data.subarray(101, 133));
+    const tokenB = new PublicKey(accountInfo.data.subarray(181, 213));
+    return clmmQuoteKey(tokenA, tokenB);
+  } else if (isRaydiumOracle(accountInfo)) {
+    const tokenA = new PublicKey(accountInfo.data.subarray(73, 105));
+    const tokenB = new PublicKey(accountInfo.data.subarray(105, 137));
+    return clmmQuoteKey(tokenA, tokenB);
+  } else {
+    return PublicKey.default;
+  }
+}
+
+function clmmQuoteKey(tokenA: PublicKey, tokenB: PublicKey): PublicKey {
+  if (
+    tokenA.equals(USDC_MINT_MAINNET) ||
+    (tokenA.equals(SOL_MINT_MAINNET) && !tokenB.equals(USDC_MINT_MAINNET))
+  ) {
+    return tokenA; // inverted
+  } else {
+    return tokenB;
+  }
+}
+// Assumes oracles.length === fallbacks.length
+export async function createFallbackOracleMap(
+  conn: Connection,
+  oracles: PublicKey[],
+  fallbacks: PublicKey[],
+): Promise<Map<PublicKey, [PublicKey, PublicKey]>> {
+  const map: Map<PublicKey, [PublicKey, PublicKey]> = new Map();
+  const accounts = await conn.getMultipleAccountsInfo(fallbacks);
```

Comment:
it's a bit sad that this is going to be re-fetched so often when it's almost never going to change